compiler/rustc_trait_selection/src/solve/search_graph.rs

```diff
@@ -201,13 +200,22 @@ impl<'tcx> SearchGraph<'tcx> {
                     },
                     available_depth,
                 )
-            {
-                inspect.goal_evaluation_kind(inspect::WipCanonicalGoalEvaluationKind::CacheHit(
-                    CacheHit::Global,
-                ));
-                self.on_cache_hit(reached_depth, encountered_overflow);
-                return result;
+            else {
+                break 'global;
+            };
+
+            if !inspect.is_noop() {
```

Comment:
Ah, I was misreading this line.

Pleas comment here saying that when we are computing a proof tree, we'll skip using the cached entry if it doesn't have a computed proof tree?